kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -131,4 +174,37 @@ protected static void validParquetColumnNames(List<String> columnNames) {
                             " ' ,;{}()\\n\\t='.", name));
         }
     }
+
+    /**
+     * Validate the supported data types. Once we start supporting additional types, take input the
+     * protocol features and validate the schema.
+     *
+     * @param dataType the data type to validate
+     */
+    protected static void validateSupportedTypes(DataType dataType) {
+        if (dataType instanceof BooleanType ||
+                dataType instanceof ByteType ||
+                dataType instanceof ShortType ||
+                dataType instanceof IntegerType ||
+                dataType instanceof LongType ||
+                dataType instanceof FloatType ||
+                dataType instanceof DoubleType ||
+                dataType instanceof DecimalType ||
+                dataType instanceof StringType ||
+                dataType instanceof BinaryType ||
+                dataType instanceof DateType ||
+                dataType instanceof TimestampType) {
+        } else if (dataType instanceof StructType) {
+            ((StructType) dataType).fields().
+                    forEach(field -> validateSupportedTypes(field.getDataType()));
+        } else if (dataType instanceof ArrayType) {
+            validateSupportedTypes(((ArrayType) dataType).getElementType());
+        } else if (dataType instanceof MapType) {
+            validateSupportedTypes(((MapType) dataType).getKeyType());
+            validateSupportedTypes(((MapType) dataType).getValueType());
+        } else {
+            throw new IllegalArgumentException("Unsupported data type: " + dataType);
```

Comment:
User facing exception in the future (due to protocol feature misalignment)? I'm good with a comment here to make sure we update it once we add that stuff

Reply:
Updated all of them to use a method in `DeltaErrors`